cloudvolume/cloudvolume.py

```diff
@@ -160,6 +157,27 @@ def __init__(self, cloudpath, mip=0, bounded=True, autocrop=False, fill_missing=
 
     self.pid = os.getpid()
 
+  @classmethod
+  def from_numpy(cls, arr, vol_path='file://tmp/'+generate_random_string(),
+                  resolution=(4,4,40), voxel_offset=(0,0,0), 
+                  chunk_size=(64,64,64)):
+    mkdir(vol_path)
+    if arr.dtype == np.uint8 or arr.dtype == np.float32:
+      layer_type = 'image'
+    elif arr.dtype == np.uint32 or arr.dtype == np.uint64:
+      layer_type = 'segmentation'
+
+    # cloud volume use fortran order, so we need to transpose 
+    # arr = np.transpose(arr)
+    # import pdb; pdb.set_trace()
+    info = cls.create_new_info(1, layer_type, arr.dtype.name, 'raw', resolution, voxel_offset, arr.shape)
```

Comment:
`num_channels=1` (I couldn't remember what it meant.)

Reply:
em, if we want to support multiple channel, I should make it more general.

Reply:
I have make it a little bit more smarter.